Adjust requirements for invalid SvcParamValues (#319)
MikeBishop/dns-alt-svc: draft-ietf-dnsop-svcb-https.md

## General requirements

Recursive resolvers MUST be able to convey SVCB records with unrecognized
SvcParamKeys, and MAY treat the entire
SvcParams portion of the record as opaque.  No part of this specification requires
recursive resolvers to alter their behavior based on its contents, even if the contents

> are invalid.  To ensure compatibility with complex SvcParam specifications,

Reviewer: ```suggestion
are invalid.  To ensure compatibility with complex and/or future SvcParam specifications,
```

Author: Why isn't this covered by the preceding requirement to pass unrecognized keys?

Reviewer: I guess it mostly is, but if we're adding justification does it hurt to remind people over and over?

Author: I don't think it's relevant to this sentence, unless you're imagining changes that formally alter the range of allowed values for an existing key.

Reviewer: An example might be future alpn values.  Perhaps there's a better way to work?

Author: "evolving"?

Reviewer 2: This spec identifies ALPN IDs as "a sequence of 1-255 octets", and in no way implies that anyone should restrict validation to currently defined identifiers.  If some future ALPN ID is more than 255 octets, there are larger problems, in that the existing parameter format wouldn't be able to encode them.

Reviewer 3: (TLS uses a 1-byte length prefix, so it's limited to 255 octets there too.)

## General requirements

Recursive resolvers MUST be able to convey SVCB records with unrecognized
SvcParamKeys, and MAY treat the entire
SvcParams portion of the record as opaque.  No part of this specification requires
recursive resolvers to alter their behavior based on its contents, even if the contents
are invalid.  Recursive resolvers MAY validate the values of recognized

> SvcParamKeys and reject records containing invalid values.  However, for
> complex value types whose interpretation might differ between implementations
> (e.g. URIs), resolvers SHOULD limit validation to basic sanity checks.

Reviewer: We should use something more descriptive/specific here.  
This wording is confusing.  (The term "sanity checks"
is also one that should avoid using from an inclusive
language perspective, if better and more descriptive terms
are available.)

Perhaps:

```suggestion
SvcParamKeys and reject records containing values 
which are invalid according to the SvsParam specification.
For complex value types whose interpretation might differ 
between implementations or have additional future
allowed values added (e.g. URIs or "alpn"), resolvers 
SHOULD limit validation to specified constraints.
```

## General requirements

Recursive resolvers MUST be able to convey SVCB records with unrecognized
SvcParamKeys, and MAY treat the entire
SvcParams portion of the record as opaque.  No part of this specification requires
recursive resolvers to alter their behavior based on its contents, even if the contents
are invalid.  Recursive resolvers MAY validate the values of recognized
SvcParamKeys and reject records containing values 

> which are invalid according to the SvsParam specification.

Reviewer: ```suggestion
which are invalid according to the SvcParam specification.
```
